### PR TITLE
RFC: Array methods

### PR DESCRIPTION
**Pull request by mjfwebb**
# Description

Adds an RFC for array methods

### CODE REVIEW

**Review by tbwiss — COMMENTED** [review]
This looks pretty good @mjfwebb!
Added one thing that I thought of.

File: docs/rfcs/rfc-020-array-methods.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    info {
+      poppedElements: ["Emil"]
+    }
+  }
+}
+```
+
+We could perform a PUSH operation like so, where the update parameter is an array containing the values to be pushed to the end of the array:
+
+```
```
Comment (MacondoExpress) [inline docs/rfcs/rfc-020-array-methods.md:52]:
```suggestion
```graphql
```
Just the formatter/prettifier :)

File: docs/rfcs/rfc-020-array-methods.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      middleNames
+    }
+  }
+}
+```
+
+### Technical considerations
+
+In an update mutation when the POP method is not used, the `info` object will contain the field `poppedElements` with a value of `[]`.
+
```
Comment (MacondoExpress) [inline docs/rfcs/rfc-020-array-methods.md:69]: A couple of things that I would add as technical considerations are:

- What happens when used on an optional/not defined field.
- What happens when the same query contains another operator/setter for the same field(I would suggest raising an error there!)

**Comment by MacondoExpress** [comment]
Added a couple of comments, but it looks good 🎉 !

File: docs/rfcs/rfc-020-array-methods.md (hunk 2 above)
Comment (mjfwebb) [reply docs/rfcs/rfc-020-array-methods.md:69]: I think I've now addressed your points here @MacondoExpress. If you agree I'll go ahead and resolve this comment.

Comment (MacondoExpress) [reply docs/rfcs/rfc-020-array-methods.md:69]: Absolutely! good joob!

**Comment by tbwiss** [comment]
IMHO this looks like a good version of the RFC to get started with the implementation!
Let's discuss this later today and also keep this PR open for visibility. We can also get rid of the draft state of this PR.

**Review by MacondoExpress — DISMISSED** [review]
LGTM 🎉 !

**Comment by angrykoala** [comment]
Considering the feature is already in progress/finished, maybe we should merge this

**Review by MacondoExpress — APPROVED** [review]

**Pull request merged**